app/src/main/java/com/example/antitheft/structure/LocationParser.java

```diff
@@ -0,0 +1,137 @@
+package com.example.antitheft.structure;
+
+import android.location.Location;
+
+import androidx.annotation.NonNull;
+
+/**
+ * This class retrieves the coordinates of a certain {@link android.location.Location} , and has
+ * a custom toString method.
+ *
+ * @author Francesco Bau'
+ * @version 0.1
+ * @since 26/02/2020
+ */
+public class LocationParser {
+    private static final String UNKNOWN_LOCATION_MESSAGE = "Location is NOT acquired.";
+    protected static final double DEFAULT_LOCATION = 0;
+    static final String DEFAULT_PROVIDER = "Default";
```

Comment:
Good point here.
I'm gonna explain it right now...
- `DEFAULT_LOCATION` is supposed to be protected, because it might be useful to subclasses, as well. Though I'm gonna change it a bit, you made me realize something.
- `DEFAULT_PROVIDER` is a local thing. It was supposed to be used in `LocationManager` class as well, but I thought to put it package-private on purpose, so only the classes inside "structure" are able to access it. I've done this, because I wasn't thinking to make `DEFAULT_PROVIDER` accessible to subclasses as well...
The main question is: what if subclasses want to put a different `DEFAULT_PROVIDER` ?
That's why I decided this way...
But I'm not 100% sure if that's the right thing... What do you think about it?

Reply:
In the end, I decided to delete `DEFAULT_LOCATION`.
But I still haven't edited `DEFAULT_PROVIDER`.
What do you think about it?

Reply:
Maybe I explained myself wrong, I didn't mean to delete the variables but only to add private etc., for example
```suggestion
    private static final String DEFAULT_PROVIDER = "Default";
```

I think it's a good idea to keep the provider saved somewhere.

Reply:
No, I get what do you mean, and yes, I wanted it to be a local thing, accessible to the classes in package `structure` only, but, if you think it's better to put it `protected` instead of `package-private`, well... I trust your opinion.
I'm gonna edit it, then....

And, I've deleted DEFAULT_LOCATION, because I could retrieve default Latitude and Longitude by doing `new Location().getLatitude()` and `new Location().getLongitude()` , so the constant was redundant.